### PR TITLE
Fix NSG e2e error

### PR DESCRIPTION
**Pull request by nwnt**
Original discussion thread:
https://redhat-external.slack.com/archives/C03F6AA3HDH/p1689891751922919

### Which issue this PR addresses:

Fixes: the spring 239 release failure, ARO-3293

It also includes the changes reverted by #3058

### What this PR does / why we need it:

Fix the subtle error found during the e2e test in Dev release.  The error occurred when the new NSG validate function is called for the FP before the flag alternation in `checkPreconfiguredNSG` called by validateSubnets.  In e2e, the subnets will not have any NSG, but the new NSG validation accesses it.  That's why it's giving a nil pointer reference error.

### Test plan for issue:

e2e and CI are now passed.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: pkg/validate/openshiftcluster_validatedynamic.go [hunk 1]
```diff
@@ -263,6 +258,11 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 		return err
 	}
 
+	err = fpDynamic.ValidatePreConfiguredNSGs(ctx, dv.oc, subnets)
```
Comment (bennerv) [inline pkg/validate/openshiftcluster_validatedynamic.go:261]: The goal was to group the specific types of validation together.  We should try to keep the service-principal validations together as much as possible.  I would rather move the SP validations to the top than move a singular FP validation to the bottom.

**Review by bennerv — CHANGES_REQUESTED** [review]
Can we elaborate on the issue, why it's an issue, and what this PR does to fix it?

File: pkg/validate/openshiftcluster_validatedynamic.go (hunk 1 above)
Comment (nwnt) [reply pkg/validate/openshiftcluster_validatedynamic.go]: Agreed.  I added more description on how the error currently occurs (TLDR: any calls to `ValidatePreconfiguredNSG` needs to happen after `ValidateSubnets`.  This should no longer be the case after GA as the logic in `ValidateSubnets` will go away).  

I moved all the SP validations to the top and FP's lower to keep them grouped.

**Review by bennerv — DISMISSED** [review]
the e2e test `ARO Operator - Azure Subnet Reconciler` needs updating to detect if it's a preconfigured NSG cluster.  If so, we should skip the e2e test for reconciling the NSG.

**Review by s-amann — DISMISSED** [review]

**Review by petrkotas — DISMISSED** [review]
Thanks @nwnt for the update. It seems all is addressed now.

**Review by SudoBrendan — DISMISSED** [review]

**Review by petrkotas — APPROVED** [review]
@nwnt thank you for the updates, I believe all comments are addressed now.

File: test/e2e/operator.go [hunk 2]
```diff
@@ -342,6 +342,15 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 	}
 
 	BeforeEach(func(ctx context.Context) {
+		// TODO remove this when GA
```
Comment (petrkotas) [inline test/e2e/operator.go:345]: @bennerv I think this is what you were asking for.

**Pull request merged**